Makefile.common

```diff
@@ -6,87 +6,142 @@ CONFIG_ENABLE_FCS16 ?= y
-    CPPFLAGS += -DTINY_HD_DEBUG=1
-    ENABLE_DEBUG=y
+	CPPFLAGS += -DTINY_HD_DEBUG=1
+	ENABLE_DEBUG=y
 endif
 ifeq ($(ENABLE_FD_LOGS),y)
-    CPPFLAGS += -DTINY_FD_DEBUG=1
-    ENABLE_DEBUG=y
+	CPPFLAGS += -DTINY_FD_DEBUG=1
+	ENABLE_DEBUG=y
 endif
 ifeq ($(ENABLE_DEBUG),y)
-    CPPFLAGS += -DTINY_DEBUG=1
+	CPPFLAGS += -DTINY_DEBUG=1
 endif
 ifeq ($(CONFIG_ENABLE_FCS32),y)
-    CPPFLAGS += -DCONFIG_ENABLE_FCS32
+	CPPFLAGS += -DCONFIG_ENABLE_FCS32
 endif
 
 ifeq ($(CONFIG_ENABLE_FCS16),y)
-    CPPFLAGS += -DCONFIG_ENABLE_FCS16
+	CPPFLAGS += -DCONFIG_ENABLE_FCS16
 endif
 
 ifeq ($(CONFIG_ENABLE_CHECKSUM),y)
-    CPPFLAGS += -DCONFIG_ENABLE_CHECKSUM
+	CPPFLAGS += -DCONFIG_ENABLE_CHECKSUM
 endif
 
 ifeq ($(CONFIG_ENABLE_STATS),y)
-    CPPFLAGS += -DCONFIG_ENABLE_STATS
+	CPPFLAGS += -DCONFIG_ENABLE_STATS
 endif
 
 .PHONY: clean library all install docs release
 
 ####################### Compiling library #########################
 
-TARGET_LIB = libtinyprotocol.a
+TARGET_LIB ?= libtinyprotocol.a
 
 OBJ_LIB += \
-        src/proto/crc/crc.o \
-        src/proto/light/tiny_light.o \
-        src/proto/hdlc/tiny_hdlc.o \
-        src/proto/fd/tiny_fd.o \
-        src/proto/half_duplex/tiny_hd.o \
-        src/proto/hal/tiny_list.o \
-        src/proto/hal/tiny_types.o \
-        src/proto/half_duplex/tiny_rq_pool.o \
-        src/TinyProtocolHd.o \
-        src/TinyProtocolFd.o \
-	src/TinyLightProtocol.o \
+		src/proto/crc/crc.o \
+		src/proto/light/tiny_light.o \
+		src/proto/hdlc/tiny_hdlc.o \
+		src/proto/fd/tiny_fd.o \
+		src/proto/half_duplex/tiny_hd.o \
+		src/proto/hal/tiny_list.o \
+		src/proto/hal/tiny_types.o \
+		src/proto/half_duplex/tiny_rq_pool.o \
+		src/TinyProtocolHd.o \
+		src/TinyProtocolFd.o \
+		src/TinyLightProtocol.o \
+
+prep:
+ifdef CONFIG_FOR_WINDOWS_BUILD
+	mkdir $(BLD)\lib $(BLD)\include
+else
+	mkdir -p $(BLD)/lib $(BLD)/include
+endif
 
 library: $(OBJ_LIB)
-	mkdir -p $(BLD)
-	$(AR) rcs $(BLD)/$(TARGET_LIB) $^
+	$(AR) rcs $(BLD)/lib/$(TARGET_LIB) $^
+
+copy_include:
+ifdef CONFIG_FOR_WINDOWS_BUILD
+	copy src\proto\crc\crc.h $(BLD)\include\ /y &\
+	copy src\proto\fd\tiny_fd.h $(BLD)\include\ /y &\
+	copy src\proto\fd\tiny_fd_int.h $(BLD)\include\ /y &\
+	copy src\proto\hal\tiny_list.h $(BLD)\include\ /y &\
+	copy src\proto\hal\tiny_types.h $(BLD)\include\ /y &\
+	copy src\proto\hal\tiny_debug.h $(BLD)\include\ /y &\
+	copy src\proto\half_duplex\tiny_hd.h $(BLD)\include\ /y &\
+	copy src\proto\half_duplex\tiny_rq_pool.h $(BLD)\include\ /y &\
+	copy src\proto\hdlc\tiny_hdlc.h $(BLD)\include\ /y &\
+	copy src\proto\light\tiny_light.h $(BLD)\include\ /y &\
+	copy src\TinyProtocolHd.h $(BLD)\include\ /y &\
+	copy src\TinyProtocolFd.h $(BLD)\include\ /y &\
+	copy src\TinyLightProtocol.h $(BLD)\include\ /y &\
+	copy src\TinyPacket.h $(BLD)\include\ /y &\
+	copy src\TinyProtocol.h $(BLD)\include\ /y 
+else
+	cp -rf src/proto/crc/crc.h \
+	src/proto/fd/tiny_fd.h \
+	src/proto/fd/tiny_fd_int.h \
+	src/proto/hal/tiny_list.h \
```

Comment:
To avoid manual support for files list and directories structure in Makefile, maybe it is simpler to use something like this:
```.make
cd src && find ./ -name \*.h -exec cp --parents {} $(BLD)/include/ \; && cd ..
```

I don't know if it works for mingw, but it definitely works for Linux

Reply:
Like this way. It does not work under mingw but I will find windows equivalent.